### PR TITLE
Add TemplateRegistryAwareInterface

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this change respect BC.

## Changelog

```markdown
### Added
-  `Sonata\AdminBundle\Templating\TemplateRegistryAwareInterface` with `getTemplateRegistry()`, `hasTemplateRegistry()` and `setTemplateRegistry()` methods
```

### CODE REVIEW

File: src/Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -3265,10 +3265,7 @@ public function canAccessObject($action, $object)
         return $this->hasAccess($action, $object);
     }
 
-    /**
-     * @return MutableTemplateRegistryInterface
-     */
-    final public function getTemplateRegistry()
+    final public function getTemplateRegistry(): ?MutableTemplateRegistryInterface
```
Comment (OskarStark) [inline src/Admin/AbstractAdmin.php:3268]: Why it can return null?

File: src/Templating/TemplateRegistryOwnerInterface.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * @author Wojciech Błoszyk <wbloszyk@gmail.com>
+ *
+ * @method MutableTemplateRegistryInterface getTemplateRegistry()
+ * @method bool                             hasTemplateRegistry()
+ * @method void                             setTemplateRegistry(MutableTemplateRegistryInterface $templateRegistry)
+ */
+interface TemplateRegistryOwnerInterface
+{
+    // NEXT_MAJOR: uncomment this method in 4.0
+    //public function getTemplateRegistry(): MutableTemplateRegistryInterface;
```
Comment (OskarStark) [inline src/Templating/TemplateRegistryOwnerInterface.php:27]: And here it cannot?

File: src/Admin/AbstractAdmin.php [hunk 3]
```diff
@@ -1353,7 +1353,7 @@ public function generateMenuUrl($name, array $parameters = [], $referenceType =
         return $this->routeGenerator->generateMenuUrl($this, $name, $parameters, $referenceType);
     }
 
-    final public function setTemplateRegistry(MutableTemplateRegistryInterface $templateRegistry)
+    final public function setTemplateRegistry(MutableTemplateRegistryInterface $templateRegistry): void
```
Comment (OskarStark) [inline src/Admin/AbstractAdmin.php:1356]: Imo this is a BC break

File: src/Admin/AbstractAdmin.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Admin/AbstractAdmin.php:3268]: I can in 3.x.

It won't in master. A next major would be needed indeed.

File: src/Admin/AbstractAdmin.php (hunk 3 above)
Comment (VincentLanglet) [reply src/Admin/AbstractAdmin.php:1356]: it's final

**Comment by phansys** [comment]
Other packages use the term "aware" instead "owner" for cases like this (`ContainerAwareInterface`, `LoggerAwareInterface`, etc).
IMO, based on that premise, we should rename this interface to `TemplateRegistryAwareInterface`.

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
@wbloszyk Please wait for multiple approval before merging (generally 2 is fine).
And if possible, it's generally better to not merge our own PR. ;)

**Comment by wbloszyk** [comment]
I ask about merge on slack. I prefer 1 approve per refactoring PR and 2+ for new feature.

File: src/Templating/TemplateRegistryAwareInterface.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Templating;
+
+/**
+ * /**
```
Comment (VincentLanglet) [inline src/Templating/TemplateRegistryAwareInterface.php:17]: Typo in comment

**Comment by VincentLanglet** [comment]
> I ask about merge on slack. I prefer 1 approve per refactoring PR and 2+ for new feature.

It's a team project. And people can't be connected every day.

**Comment by greg0ire** [comment]
Also, it's open-source software… there are a loooooot more users than a company project.

**Pull request merged**